src/decoder/decodeqrdata.ts

```diff
@@ -219,7 +219,7 @@ function decodeByteSegment(bits: BitStream, result: resultByteArray, count: numb
   for (var i = 0; i < count; i++) {
     readBytes[i] = bits.readBits(8);
   }
-  result.val = readBytes
+  Array.prototype.push.apply(result.val, readBytes);
```

Comment:
Why do we need to do `Array.prototype.push`? Can we not use `result.val.push(readBytes)` here?